dashboard/test/controllers/application_controller_test.rb

```diff
@@ -63,21 +63,50 @@ class ApplicationControllerTest < ActionDispatch::IntegrationTest
       end
     end
 
-    context 'when user is not lockable' do
-      let(:user_lockable?) {false}
+    context 'when user is locked out' do
+      let(:user_is_locked_out?) {false}
 
-      it 'does not lock out user' do
-        Services::ChildAccount.expects(:lock_out).with(user).never
+      it 'does not redirect to lockout page' do
         get root_path
+        refute_redirect_to lockout_path
       end
     end
 
     context 'when no CPA experience' do
       let(:cpa_experience_phase) {nil}
 
-      it 'does not lock out user' do
-        Services::ChildAccount.expects(:lock_out).with(user).never
+      before do
+        expect_grace_period_handler_call.never
+        expect_lockout_handler_call.never
+      end
+
+      it 'does not redirect to lockout page' do
         get root_path
+        refute_redirect_to lockout_path
+      end
+    end
+
+    context 'when error is raised during locking out' do
+      let(:error) {StandardError.new('expected_error')}
+
+      before do
+        expect_lockout_handler_call.never
+        Services::ChildAccount::LockoutHandler.expects(:call).with(user: user).raises(error)
```

Comment:
```suggestion
        expect_lockout_handler_call.never
        expect_lockout_handler_call.raises(error)
```
Does this work? Is the `.never` necessary?

Reply:
Unfortunately, `expect_lockout_handler_call` cannot be stubbed to raise any errors because it's already stubbed to return a value: https://github.com/code-dot-org/code-dot-org/blob/38ff97f57d25cdd6ea6ee749d4e3c644ae019209/dashboard/test/controllers/application_controller_test.rb#L17

Reply:
The refactoring of the tests: https://github.com/code-dot-org/code-dot-org/pull/59241/commits/fca955a5a0f8474abbf51a750fd77ae3e916f25c